File: Pixly-BE/Pixly.Models/DTOs/PhotoDetail.cs
namespace Pixly.Models.DTOs
{
    public class PhotoDetail
    {
        public int PhotoId { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public string Url { get; set; }
        public string Slug { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public long FileSize { get; set; }
        public string Format { get; set; }
        public DateTime UploadedAt { get; set; }
        public string UserId { get; set; }
        public User? User { get; set; }
        public string? State { get; set; }
        public int ViewCount { get; set; }
        public int LikeCount { get; set; }
        public int DownloadCount { get; set; }
        public string Orientation { get; set; }
        public ICollection<PhotoTag> PhotoTags { get; set; } = new List<PhotoTag>();
        public bool IsCurrentUserLiked { get; set; }
        public bool IsCurrentUserSaved { get; set; }
        public int? Price { get; set; }
    }
}

File: Pixly-FE/src/app/core/models/DTOs/PhotoDetail.ts
import { PhotoTag } from "./PhotoTag";
import { User } from "./User";

export interface PhotoDetail {
    photoId: number;
    title: string;
    description: string | null;
    url: string;
    slug: string;
    width: number;
    height: number;
    fileSize: number;
    format: string;
    uploadedAt: string;
    userId: string;
    user: User | null;
    state: string | null;
    viewCount: number;
    likeCount: number;
    downloadCount: number;
    orientation: string;
    photoTags: PhotoTag[];
    isCurrentUserLiked: boolean; 
    isCurrentUserSaved: boolean;
    price: number | null;
}